### DISCUSSION TITLE
What Will Microsoft Build Using MAUI?

### DISCUSSION THREAD

**Discussion by LeeWhite187** [dotnet/maui#12019] [Ideas]
I'm a Xamarin fan, for some seven plus years, now.
And, I appreciate and respect the developers that have worked hard to provide stability, support, and contributed to the successful, cross-platform abstraction, that Xamarin has become.
So, this is not a dig at anyone on the MAUI team. I don't envy you for the "shoes" you're being expected to fill.

I have one complaint about MAUI...
Why is Microsoft NOT using MAUI in their own major projects, such as their own flagship applications?
And, please don't bother with myopic replies about tooling and platform choices being pre-existing... because, for example, Xamarin was owned by MS before MSTeams published... and MSTeams uses Electron.

The Xamarin community of users and developers have spent years of pain and risk (by choosing Xamarin over native), in using Xamarin for real projects.
Why is Microsoft choosing to not feel that same pain, with MAUI (a platform they're pushing on the same community)?
Does Microsoft expect the established community of Xamarin supporters and developers to spend ANOTHER round of several years working through, and feeling the pain of, a new set of instabilities and bugs (in MAUI), and deal with less functionality than Xamarin already has (because of the intrinsic lag of getting pertinent libraries brought-forward to MAUI), while trying it for real projects?

I spent years following Xamarin, and welcoming the pain and risk of using it in real projects, for what it might achieve.
Xamarin proved to us, that a cross-platform framework was viable. And, we happily created a community for it.
Does Microsoft expect the community to again put real projects at risk (by mandatorily switching to MAUI (because Microsoft has or will, sunset what Xamarin depends on), all while Microsoft chooses what bugs to triage, without any "skin in the game" of actually using MAUI for real projects?

Why does Microsoft expect the Xamarin community, they depend on for MAUI support, to fully endure this second round of risk and pain for a replacement?
Is this how Microsoft fosters the established Xamarin community?

I love Xamarin for what it proved.
But, I can't rationalize how Microsoft expects the community to carry the risk and pain of trying to use MAUI in real projects, while Microsoft rides on the claims that the Xamarin community proved, but (Microsoft) doesn't actually put their own skin in the game of trying to use MAUI in their own products.

The community proved Xamarin works for real projects.
What will Microsoft build using MAUI?

**Comment by manishbisen** [comment]
Thanks for bringing this up

**Comment by Majestic-Mustang** [comment]
It's apparent that Microsoft lacks good leadership and direction when they themselves don't have the commitment or confidence to dog-food their products.

Only when Microsoft creates their flagship products using MAUI, the community will feel confident on this product. As that will prove that Microsoft will have to make MAUI successful for making their flagship products successful. 

Google already uses Flutter in their flagship apps. Why can't Microsoft do the same?

**Reply by Xyncgas** [reply]
We are developers, the technology itself should provide a taste that's recognizable by a fine man, we crave to make something good, only when the technology is terrible, we explain why

**Reply by davidbuckleyni** [reply]
Google actually droped one of their major product lines from Flutter cause couldnt get round major issues to say that flutter is a one stop shop to is ill fated as flutter has major issues on performances and the ui layer.

**Comment by Majestic-Mustang** [comment]
Can someone from Microsoft please answer this? @PureWeen @davidortinau  @jfversluis

**Comment by xtuzy** [comment]
Currently only Azure app use xamarin, maybe this app will use maui when microsoft not support xamarin. 
I open google play store, find microsoft don't have many hot app, if these app need cross platform ui framework, they will use flutter, because flutter can as a plugin in existing project, if they use maui, they need rewrite app.
Currently Teams use reactnative and electron(webview2), many people say web is future, because cpu will be more powerful, you can see web app framework like tauri is very hot, i feel maui need do more in webview+native, not only for blazor, blazor is minority.

**Reply by akhanalcs** [reply]
@xtuzy Isn't Blazor Hybrid similar to Tauri?

**Reply by xtuzy** [reply]
You don't get that i want say. web have many framework and many developer, blazor is little one and use c#, more web framework use js, i want maui not only attract dotnet developer. such as https://github.com/dotnet/maui/discussions/12009

**Reply by Xyncgas** [reply]
At least it's easy to publish MAUI to windows store, makes my application looks pretty authentic when it's on a store
I like the option to publish to executable and run it directly too, like regular apps, but UI is cross platform so is the rt that's awesome

**Reply by akhanalcs** [reply]
> You don't get that i want say. web have many framework and many developer, blazor is little one and use c#, more web framework use js, i want maui not only attract dotnet developer. such as #12009

Gotcha. Eilon posted a latest example where he uses ReactJs with MAUI which is pretty cool. Is that what you were looking for? If Yes, please give it a try.

**Comment by LeeWhite187** [comment]
I'll say this. (assuming everyone has read my previous about my long-running appreciation for the XF community).

In the 2000's, Microsoft created a compelling case for switching to .NET development. They provided great support, polished, end-to-end tooling, and included a kitchen-sink of functionalities (Granted, this was market competition against JAVA. But, it paid off).
And, their efforts created an entire ecosystem for .NET and derivative technologies, for example: Xamarin... now MAUI.

But if Microsoft is not willing to have real skin in the game, by using Xamarin/MAUI in their own products, and instead chooses 3rd-party frameworks for their own flagship products (ex: Electron, React Native, Flutter, JS, etc), then what does that signal to the .NET/Xamarin/MAUI community, and any developer or company that may consider MAUI for real projects?

And, how does Microsoft entice/compel companies (indie devs, small shops, or even the next unicorn) to consider risking reputation/money/time developing applications in a framework (MAUI), that the framework's own sponsor (Microsoft) chooses against (for its own flagship products)?

Microsoft. Show the XF community, and your own MAUI team, most importantly, that you are committed to MAUI being a viable platform choice.
What will Microsoft build using MAUI?

**Reply by Majestic-Mustang** [reply]
You're right. Microsoft lacks good leadership when they themselves don't have the confidence to use their tools. It's very disheartening to .NET devs as if they're second-class citizens in their own world.

**Reply by Majestic-Mustang** [reply]
IMHO, Satya Nadella is not the right person to drive Microsoft.
That man has no vision and is a follower than a leader.

**Reply by Xyncgas** [reply]
What a company can do, is look at these project as a good incubation process for cheap labors, pay the people who develop MAUI to develop applications using MAUI that gives you all the confidence you'll need

**Comment by davidortinau** [comment]
Today we have Microsoft apps currently using Xamarin, upgrading to .NET MAUI, and new apps starting with .NET MAUI. Are 365, Bing, MS IoT, Sharepoint, Azure, and Dynamics flagship to you? SeeingAI, Sketch 360, or Microsoft Advertising Editor? And there are others.

The current phase of release we are in is about the ecosystem transitioning (libraries and services making the jump from .NET Framework to .NET) and the SDK stabilizing (test infrastructure, devops, features, quality). There has been improvement release after release over the past 6 months since the SDK GA. Visual Studio tooling completed GA only 1 month ago.

The next phase we expect to be coming into soon will be gaining momentum building new apps, seeing more customer stories, case studies, and upgrades from Xamarin / .NET Framework.

**Reply by manishbisen** [reply]
When next phase will be coming?

**Reply by FM1973** [reply]
> When next phase will be coming?

I´d like to know this as well...

**Reply by Xyncgas** [reply]
Windows 12 can appeal to the .NET developers by making visual studio / dotnet apps run faster and better on this operating system than other operating system, I would like to be microsoft's favorite because I am dotnet developer, and microsoft can show its affirmation by forming a parasocial relationship with me providing features in the .NET rt that would work better on Windows even though it can work on other platform, providing a market that has luxury / flagship / quality mobile devices running windows so I can live off Microsoft without google or apple, even though I can publish to IOS and androids too.

The fact that microsoft has their foot in the cake would give me more access to audience then when Microsoft lift its finger I think me and many people will do anything for you

**Reply by YiannisBourkelis** [reply]
The good thing is that now the MAUI team will receive bug reports from inside that should be solved to be able to deliver those MAUI Microsoft apps. From my very recent experience, it was impossible to build a UI with MAUI without hitting bugs every day.  Layout issues, controls disappearing, inconsistent look between windows and mobile, poor collection view performance and so many more. Every developer who will try to use MAUI now, will have to deal with those issues.

**Reply by MartinRothschink** [reply]
Yes, I'm dealing with these bugs every day. Each release goes 2 steps forward and one step back. Things working in MAUI .Net 6 are broken in .Net 7. And guess what? Each bug report is always moved to the Backlog milestone. I don't need new features as long as the fundamental basics are broken. It's really frustrating...

**Reply by davidortinau** [reply]
Moving to the backlog simply means it's not part of the current sprint / service release. Based on impact prioritization any issue can come from the backlog into the next sprint. To influence that, make sure the issue has ample information to reproduce it and add any reactions to weight it. 

We will work on features when needed, but the vast majority of each sprint right now is focused on quality which is infrastructure, testing, and bug fixing. Our project boards show what's in each sprint (iteration).

**Reply by saint4eva** [reply]
> > When next phase will be coming?
> 
> I´d like to know this as well...

I thought you were asking for apps built by Microsoft using Maui?

**Reply by mphill** [reply]
MS is using React Native for or in Teams, Office and Outlook.  If MS used MAUI for those, adoption for MAUI would go viral.

It would be interesting to to see if these React Native apps could be ported to a MAUI Hybrid. 

@davidortinau to confirm are you saying https://apps.apple.com/us/app-bundle/microsoft-365-mobile-apps/id1564009682 will be ported to MAUI?

**Reply by LeeWhite187** [reply]
@davidortinau 
Well. 365 in MAUI would be a good start... of Microsoft putting their reputation on the line (to build something in MAUI).
Hopefully, when they start pushing the MAUI implementations of their own larger and more-visible applications (products whose customers demand stability) past the internal Hello Worlds and alpha stages, the XF/MAUI community will start to benefit from the increased stability (in the MAUI framework) that a Microsoft product team requires.

The XF/MAUI community will benefit from that level of support... and confidence in being able to invest in MAUI as a viable option for stable mobile apps with reliable tooling.

**Reply by manishbisen** [reply]
Maui still struggling to get stable. Less number of man power against big number of bugs is biggest concern.If Microsoft continue in same way then it will take many years to maui sdk get stable. Microsoft need to hire more developers & to speed up there development speed.This is the need of the time.

**Reply by LeeWhite187** [reply]
Don't get me wrong. I'm not trying to discount the efforts of the MAUI team. MAUI is a tremendous undertaking, as we've witnessed (and felt) through the last 7+ years of XF evolution.

But, there are two development extremes for projects/frameworks:
Prioritize sprints that maintain compatibility with the underlying library cadence (NET release schedule).
Or, prioritize sprints that fix long-standing bugs, stability problems, and feature fidelity issues.
A good balance of these two extremes is the only way for MAUI to thrive and be seen as a credible framework.
And, I would expect Microsoft leadership to realize that.

So, I question why Microsoft leadership is choosing to resource-starve MAUI, by prioritizing the efforts of a burdened team with sprints that strive for MAUI compatibility with the never-stopping, NET release cadence, versus providing adequate sponsorship and resources to solve actual issues within each release.

It's no stretch to realize that... the stability issues, bugs, and feature-fidelity misses that people post here, would be much faster-solved and benefit everyone in the community, if Microsoft heavily chose MAUI for apps that ordinary retail customers consume (reading that as: apps that require stability, function, and form).

Until that level of adoption occurs, forcing Microsoft to heavily invest in MAUI development (because they would then, actually require it to be stable and feature-rich for their own products), we're in the current status quo of voting on problems, and hoping necessary bug and stability fixes get prioritized for the next sprint, done by a team of maintainers, already fully-burdened with keeping compatibility with the Net release cycle, all while Microsoft enjoys the benefit of flying banners of market leadership in cross-platform UI and open source community stewardship.

Surely, Microsoft can see the benefit that support and shared risk would have for MAUI.

**Comment by FM1973** [comment]
@davidortinau 
Thanks for the info. The last service release is more than a month old. There are severe bugs which keep us all from releasing maui apps.
When can we expect the next release? My customer want´s to know and so do I.

**Reply by jsuarezruiz** [reply]
Was released yesterday https://github.com/dotnet/maui/releases/tag/7.0.52
